In [1]:
import pandas as pd

In [9]:
df = pd.read_excel('files/hypothesis_testing.xlsx', skiprows=[0,1,2])
del df[df.columns[0]]
del df[df.columns[-1]]

In [11]:
df.tail()

Surname       Name  Age Gender        Country  \
169  Osturnka      Adeel   41   Male  United States   
170  Punjabhi      Louis   56   Male  United States   
171    Cockel      James   40   Male  United States   
172    Gordon      David   38   Male  United States   
173     Knapp  Bradley J   40   Male  United States   

                     Ethnicity Start_date         Department  \
169                      White 2013-09-30  Production          
170                      White 2014-01-06  Production          
171                      White 2013-07-08  Production          
172                      White 2012-07-02  Production          
173  Black or African American 2014-02-17  Production          

                    Position   Salary  
169  Production Technician I  33280.0  
170  Production Technician I  33280.0  
171  Production Technician I  31200.0  
172  Production Technician I  31200.0  
173  Production Technician I  29120.0

# Test if there is a significant difference in pay between male and female

- Sample -> male; female
- Independent samples

## We will use a hypothesis test for a mean salary

### $H_0$: $\mu_m - \mu_f$ = 0
### $H_1$: $\mu_m - \mu_f$ $\ne$ 0

In [41]:
df.groupby('Gender').Salary.describe()

count          mean           std      min      25%      50%  \
Gender                                                                 
Female   98.0  65736.914286  33130.318859  29120.0  41600.0  50440.0   
Male     76.0  72300.526316  35233.955988  29120.0  41600.0  56160.0   

             75%       max  
Gender                      
Female   98696.0  166400.0  
Male    114400.0  135200.0

In [58]:
from scipy.stats import ttest_ind

male = df[df['Gender']=='Male']
female = df[df['Gender']=='Female']

ttest_ind(male['Salary'], female['Salary'], equal_var=True)

Ttest_indResult(statistic=1.2606595770038977, pvalue=0.20913896861176579)

### The T-score is 1.26 and the p-value is 0.209 >> all common levels of significance -> we cannot reject $H_0$

- We accpet $H_0$
- There is no wage gap between men and women

### What if there is a hidden wage gap?

#### Divide the data into two groups: those earning above 35k and those below:

In [72]:
male_above35 = df[(df['Gender']=='Male') & (df['Salary'] > 35000)]
male_below35 = df[(df['Gender']=='Male') & (df['Salary'] < 35000)]

female_above35 = df[(df['Gender']=='Female') & (df['Salary'] > 35000)]
female_below35 = df[(df['Gender']=='Female') & (df['Salary'] < 35000)]

In [73]:
print('Above 35K')
print(ttest_ind(male_above35['Salary'], female_above35['Salary'], equal_var=False))
print('\n-----------------------\n')

print('Below 35K')
print(ttest_ind(male_below35['Salary'], female_below35['Salary'], equal_var=False))

Above 35K
Ttest_indResult(statistic=0.9470129986806425, pvalue=0.345265315041716)

-----------------------

Below 35K
Ttest_indResult(statistic=0.33183832034331656, pvalue=0.7449571682010134)


In [75]:
female_above35

Surname      Name  Age  Gender        Country  \
0         Bold  Caroline   63  Female  United States   
1       Zamora  Jennifer   38  Female  United States   
2     Houlihan     Debra   51  Female  United States   
3     Bramante     Elisa   34  Female  United States   
4   Del Bosque     Keyla   38  Female  United States   
..         ...       ...  ...     ...            ...   
78      Nguyen  Lei-Ming   33  Female  United States   
79        Gold   Shenice   25  Female  United States   
80     Mahoney    Lauren   31  Female  United States   
81    Peterson     Kayla   44  Female  United States   
82        Rose    Ashley   43  Female  United States   

                    Ethnicity Start_date            Department  \
0                       White 2012-07-02      Executive Office   
1                       White 2010-04-10                 IT/IS   
2                       White 2014-05-05                 Sales   
3   Black or African American 2009-01-05     Production          
4   Black or African American 2012-01-09  Software Engineering   
..                        ...        ...                   ...   
78                      White 2013-07-08     Production          
79                      White 2013-11-11     Production          
80                      White 2014-01-06     Production          
81                      White 2010-04-26     Production          
82                      White 2014-01-06     Production          

                   Position    Salary  
0           President & CEO  166400.0  
1                       CIO  135200.0  
2         Director of Sales  124800.0  
3    Director of Operations  124800.0  
4         Software Engineer  118809.6  
..                      ...       ...  
78  Production Technician I   39520.0  
79  Production Technician I   35360.0  
80  Production Technician I   35360.0  
81  Production Technician I   35360.0  
82  Production Technician I   35360.0  

[83 rows x 10 columns]

### Below 35 -> no wage gap HUGE p-value
### Above 35 -> Low p-value we reject $H_0$

#### The T-score 